lib/zlib.js

```diff
@@ -545,17 +547,7 @@ function processCallback() {
                self._chunkSize); // out_len
     return;
   }
-
-  if (availInAfter > 0) {
-    // If we have more input that should be written, but we also have output
-    // space available, that means that the compression library was not
-    // interested in receiving more data, and in particular that the input
-    // stream has ended early.
-    // This applies to streams where we don't check data past the end of
-    // what was consumed; that is, everything except Gunzip/Unzip.
-    self.push(null);
-  }
```

Comment:
Removing this is a breaking change, the test added in #26363 should fail (once the linter issue has been fixed here).

Reply:
Yes, I didn't realize this had already gone out the door. Nor did I realize that I should be trapping for the `end` event and not the `finish` event! 

Thanks for the code review!